File: app/src/main/java/com/payby/pos/ecr/utils/ActivityManager.java
package com.payby.pos.ecr.utils;

import android.app.Activity;

import java.util.LinkedList;

public class ActivityManager {

    public static final LinkedList<Activity> activityList = new LinkedList<>();

    public static void pushActivity(Activity activity) {
        boolean contains = activityList.contains(activity);
        if (contains) {
            if (activityList.getLast() != activity) {
                activityList.remove(activity);
                activityList.add(activity);
            }
        } else {
            activityList.add(activity);
        }
    }

    public static void popActivity(Activity activity) {
        activityList.remove(activity);
    }

    public static void finishActivity(Activity activity) {
        if (activity.isFinishing() == false) {
            activity.finish();
        }
        activityList.remove(activity);
    }

    public static void finishAll() {
        for (Activity activity : activityList) {
            if (activity.isFinishing() == false) {
                activity.finish();
            }
        }
        activityList.clear();
    }

    public static Activity getTopActivity(){
      return activityList.getLast();
    }

}

File: infrastructure/common/src/main/java/com/payby/pos/common/helper/ActivityManager.kt
package com.payby.pos.common.helper

import android.app.Activity
import java.util.LinkedList

object ActivityManager {

    private val activityList = LinkedList<Activity>()

    /**
     * push the specified [activity] into the list
     */
    fun pushActivity(activity: Activity) {
        val contains = activityList.contains(activity)
        if (contains) {
            if (activityList.last != activity) {
                activityList.remove(activity)
                activityList.add(activity)
            }
        } else {
            activityList.add(activity)
        }
    }

    /**
     * pop the specified [activity] into the list
     */
    fun popActivity(activity: Activity) {
        activityList.remove(activity)
    }

    fun finishActivity(activity: Activity) {
        activityList.remove(activity)
        activity.finish()
    }

    fun finishAllActivities(packageName: String) {
        val iterator = activityList.iterator()
        var has = iterator.hasNext()
        while (has) {
            val activity = iterator.next()
            val bool = activity.javaClass.name.startsWith(packageName)
            if (bool) {
                iterator.remove()
                activity.finish()
            }
            has = iterator.hasNext()
        }
    }

    fun finishAllActivities() {
        for (activity in activityList) {
            activity.finish()
        }
        activityList.clear()
    }

}